src/serde/deserialize_program.rs

```diff
@@ -172,13 +172,21 @@ pub fn deserialize_program_json(path: &Path) -> Result<ProgramJson, ProgramError
     Ok(program_json)
 }
 
-pub fn deserialize_program(path: &Path) -> Result<Program, ProgramError> {
+pub fn deserialize_program(path: &Path, entrypoint: &str) -> Result<Program, ProgramError> {
     let program_json: ProgramJson = deserialize_program_json(path)?;
+
+    let main_pc = match program_json
```

Comment:
For consistency this could be named `entrypoint_pc`.

Reply:
+!